### PR TITLE
refactor: refactored mainwindow

### PR DESCRIPTION
**Pull request by uyraq2001**

### CODE REVIEW

File: src/app/mainwindow.cpp [hunk 1]
```diff
@@ -23,64 +23,89 @@
...
 namespace ab
 {
+class MainWindowPrivate{
+public:
+    MainWindowPrivate(MainWindow *window)
+        : ui(new Ui::MainWindow)
+        , quitShortcut(new QShortcut(QKeySequence(Qt::CTRL + Qt::Key_Q), window, SLOT(close())))
+        , model(nullptr)
+        , controller(nullptr)
+        , settings(std::make_unique<MainWindowSettings>(window, ui)){}
```
Comment (mchernigin) [inline src/app/mainwindow.cpp:33]: Not formatted correctly for some reason.

File: src/app/mainwindow.cpp [hunk 2]
```diff
@@ -23,64 +23,89 @@
...
+
+    ~MainWindowPrivate()
+    {
+        delete ui;
+        controller.release();
+        model.release();
```
Comment (mchernigin) [inline src/app/mainwindow.cpp:39]: Smart pointers deallocate memory in the destructor. If you call release, you are taking "ownership" responsibility to deallocate memory yourself. But in this case you don't.

This 2 lines should not be here.

File: src/app/mainwindow.cpp [hunk 3]
```diff
@@ -23,64 +23,89 @@
...
+
+    Ui::MainWindow *ui = nullptr;
+    QShortcut *quitShortcut = nullptr;
+    std::unique_ptr<QStandardItemModel> model = nullptr;
+    std::unique_ptr<Controller> controller = nullptr;
+    std::unique_ptr<MainWindowSettings> settings = nullptr;
+};
+
 MainWindow::MainWindow(QWidget *parent)
     : QMainWindow(parent)
```
Comment (mchernigin) [inline src/app/mainwindow.cpp:55]: Do we actually need to initialize QMainWindow if we MainWindow is derived from QMainWinow?

**Review by mchernigin — CHANGES_REQUESTED** [review]

File: src/app/mainwindow.cpp (hunk 2 above)
Comment (mchernigin) [reply src/app/mainwindow.cpp:39]: If you don't actually "own" underlying data, you should not use `unique_ptr` here.

File: src/app/controller.cpp [hunk 4]
```diff
@@ -32,12 +39,14 @@ const QString DBUS_LOCAL_APP_GET_DESKTOP_FILE  = "info";
 
 Controller::Controller(MainWindow *w, std::unique_ptr<model::Model> m, QObject *parent)
     : QObject{parent}
-    , window(w)
-    , model(std::move(m))
+    , d(new ControllerPrivate)
```
Comment (AlexSP0) [inline src/app/controller.cpp:42]: How long does ControllerPrivate live?

**Review by AlexSP0 — CHANGES_REQUESTED** [review]

**Review by mchernigin — APPROVED** [review]

**Review by AlexSP0 — APPROVED** [review]

**Pull request merged**